internal/expiration/handler.go

```diff
@@ -0,0 +1,154 @@
+package expiration
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	"github.com/kyma-project/kyma-environment-broker/common/orchestration"
+	"github.com/kyma-project/kyma-environment-broker/internal"
+	"github.com/kyma-project/kyma-environment-broker/internal/broker"
+	"github.com/kyma-project/kyma-environment-broker/internal/httputil"
+	"github.com/kyma-project/kyma-environment-broker/internal/ptr"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage/dberr"
+	"github.com/kyma-project/kyma-environment-broker/internal/suspension"
+	"github.com/pivotal-cf/brokerapi/v8/domain"
+	"github.com/sirupsen/logrus"
+)
+
+type Handler interface {
+	AttachRoutes(router *mux.Router)
+}
+
+type handler struct {
+	instances           storage.Instances
+	operations          storage.Operations
+	deprovisioningQueue suspension.Adder
+	log                 logrus.FieldLogger
+}
+
+func NewHandler(instancesStorage storage.Instances, operationsStorage storage.Operations, deprovisioningQueue suspension.Adder, log logrus.FieldLogger) Handler {
+	return &handler{
+		instances:           instancesStorage,
+		operations:          operationsStorage,
+		deprovisioningQueue: deprovisioningQueue,
+		log:                 log.WithField("service", "ExpirationEndpoint"),
+	}
+}
+
+func (h *handler) AttachRoutes(router *mux.Router) {
+	router.HandleFunc("/expire/service_instance/{instance_id}", h.expireInstance).Methods("PUT")
+}
+
+func (h *handler) expireInstance(w http.ResponseWriter, req *http.Request) {
+	instanceID := mux.Vars(req)["instance_id"]
+
+	h.log.Info("Expiration triggered for instanceID: ", instanceID)
+	logger := h.log.WithField("instanceID", instanceID)
```

Comment:
The object gets logger from main func which already has some populated context fields common for every KEB operation. We use logger from main to create a new logger with one additional context field `instanceID` which must be different for every expiration request. We should keep extra info just for the expiration process. If we would use the same, saved in handler logger from main, then it would have not relevant info for every other operation in KEB.